ziti/edge/messages.go

```diff
@@ -350,6 +358,125 @@ func NewDialFailedMsg(connId uint32, message string) *channel.Message {
+			Type: &edge_client_pb.PostureResponse_Macs_{
+				Macs: &edge_client_pb.PostureResponse_Macs{
+					Addresses: typedResp.MacAddresses,
+				},
+			},
+		})
+	case *rest_model.PostureResponseProcessCreate:
+		protoResponses = append(protoResponses, &edge_client_pb.PostureResponse{
+			Type: &edge_client_pb.PostureResponse_ProcessList_{
+				ProcessList: &edge_client_pb.PostureResponse_ProcessList{
+					Processes: []*edge_client_pb.PostureResponse_Process{
+						{
+							Path:               typedResp.Path,
+							IsRunning:          typedResp.IsRunning,
+							Hash:               typedResp.Hash,
+							SignerFingerprints: typedResp.SignerFingerprints,
+						},
+					},
+				},
+			},
+		})
+	case *rest_model.PostureResponseOperatingSystemCreate:
+		protoResponses = append(protoResponses, &edge_client_pb.PostureResponse{
+			Type: &edge_client_pb.PostureResponse_Os{
+				Os: &edge_client_pb.PostureResponse_OperatingSystem{
+					Type:    stringz.OrEmpty(typedResp.Type),
+					Version: stringz.OrEmpty(typedResp.Version),
+					Build:   typedResp.Build,
+				},
+			},
+		})
+	case *rest_model.PostureResponseEndpointStateCreate:
+		if typedResp.Woken {
+			protoResponses = append(protoResponses, &edge_client_pb.PostureResponse{
+				Type: &edge_client_pb.PostureResponse_Woken_{
+					Woken: &edge_client_pb.PostureResponse_Woken{
+						Time: timestamppb.Now(),
+					},
+				},
+			})
+		}
+		if typedResp.Unlocked {
+			protoResponses = append(protoResponses, &edge_client_pb.PostureResponse{
+				Type: &edge_client_pb.PostureResponse_Unlocked_{
+					Unlocked: &edge_client_pb.PostureResponse_Unlocked{
+						Time: timestamppb.Now(),
+					},
+				},
+			})
+		}
+	case *PostureResponseTotp:
+		protoResponses = append(protoResponses, &edge_client_pb.PostureResponse{
+			Type: &edge_client_pb.PostureResponse_TotpToken_{
+				TotpToken: &edge_client_pb.PostureResponse_TotpToken{
+					Token: typedResp.TotpToken,
+				},
+			},
+		})
+	}
+
+	var messages []*channel.Message
+	for _, protoResponse := range protoResponses {
+		b, _ := proto.Marshal(protoResponse)
+		message := channel.NewMessage(ContentTypePostureResponse, b)
+		messages = append(messages, message)
+	}
+	return messages
+}
+
+var _ rest_model.PostureResponseCreate = (*PostureResponseTotp)(nil)
+
+// PostureCheckTypeTOTP mimics rest_api.PostureCheckType*
+const PostureCheckTypeTOTP rest_model.PostureCheckType = "totp"
+
+// PostureResponseTotp is a posture response that contains a TOTP token. Used to track the last time
+// a client has performed TOTP code submission. This exists outside of the REST APIs as it is only
+// needed in HA systems and is not used for legacy API Sessions. It does conform to the REST APIs generated
+// interfaces in order for it to be included in interface typed slices.
+type PostureResponseTotp struct {
+	TotpToken string //should be a signed, API Session scoped JWT. See common.TotpClaims
+	Id        *string
+}
+
```

Comment:
Should probably be using pointer receivers, see staticcheck error below

Reply:
You are correct. Fixed.